api/src/main/java/com/epam/pipeline/manager/datastorage/providers/azure/AzureHelper.java

```diff
@@ -1,51 +1,103 @@
+    public static AzureResourceManager buildClient(final AzureCredentials credentials) throws AuthenticationException {
         try {
-            final Azure.Configurable builder = Azure.configure()
-                    .withLogLevel(LogLevel.BASIC);
-            return authenticate(authFile, builder)
+            final AzureResourceManager azure = AzureResourceManager
+                    .authenticate(credentials.getCredential(), credentials.getProfile())
                     .withDefaultSubscription();
-        } catch (IOException e) {
-            log.error(e.getMessage(), e);
-            throw new AzureException(e);
+            log.info("Authenticated to subscription: {}", azure.subscriptionId());
+            return azure;
+        } catch (ManagementException e) {
+            throw new AuthenticationException(String.format("Failed to authenticate to Azure: %s", e.getMessage()));
         }
     }
 
-    private static Azure.Authenticated authenticate(final String authFile,
-                                             Azure.Configurable builder) throws IOException {
-        if (StringUtils.isBlank(authFile)) {
-            return builder.authenticate(getAzureCliCredentials());
+    public static AzureCredentials getAzureCredentials(final AzureRegion region) throws AuthenticationException {
+        TokenCredential credential;
+        String tenantId;
+        String subscriptionId;
+        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+        if (StringUtils.isNotBlank(region.getManagedIdentity())) {
+            log.info("Authentication using Managed Identity");
+            credential = new DefaultAzureCredentialBuilder()
+                    .managedIdentityClientId(region.getManagedIdentity())
+                    .build();
+        } else if (StringUtils.isNotBlank(region.getAuthFile())) {
+            log.info("Authentication using Azure auth file");
+            final File authFile = new File(region.getAuthFile());
+            final Map<String, String> config;
+            try {
+                config = new ObjectMapper().readValue(authFile, Map.class);
+            } catch (IOException e) {
+                throw new AuthenticationException("Failed to get Azure credentials from auth file.");
+            }
+            final String clientId = config.get(CLIENT_ID);
+            final String clientSecret = config.get(CLIENT_SECRET);
+            tenantId = config.get(TENANT_ID);
+            subscriptionId = config.get(SUBSCRIPTION_ID);
+            credential = new ClientSecretCredentialBuilder()
+                    .clientId(clientId)
+                    .clientSecret(clientSecret)
+                    .tenantId(tenantId)
+                    .build();
+            profile = new AzureProfile(tenantId, subscriptionId, AzureEnvironment.AZURE);
+        } else {
+            throw new AuthenticationException("Failed to get Azure credentials.");
```

Comment:
Can you expand message and log that no values of auth file or menagedIdentity where provided